networkx/drawing/nx_pylab.py

```diff
@@ -890,6 +891,207 @@ def _connectionstyle(posA, posB, *args, **kwargs):
+
+    arrowstyle : str (default='-\|>' for directed graphs)
+        For directed graphs defaults to '-\|>',
+        For undirected graphs default to '-'.
+
+        See `matplotlib.patches.ArrowStyle` for more options.
+
+    connectionstyle : string (default="arc3")
+        For multiedges, connectionstyle is overriden by the method. For other edges,
+        pass the connectionstyle parameter to create curved arc of rounding
+        radius rad. For example, connectionstyle='arc3,rad=0.2'.
+        See `matplotlib.patches.ConnectionStyle` and
+        `matplotlib.patches.FancyArrowPatch` for more info.
+
+    arrowsize : int (default=10)
+        For directed graphs, choose the size of the arrow head's length and
+        width. See `matplotlib.patches.FancyArrowPatch` for attribute
+        `mutation_scale` for more info.
+
+    node_size : scalar or array (default=300)
+        Size of nodes. Though the nodes are not drawn with this function, the
+        node size is used in determining edge positioning.
+
+    nodelist : list, optional (default=G.nodes())
+       This provides the node order for the `node_size` array (if it is an array).
+
+    node_shape :  string (default='o')
+        The marker used for nodes, used in determining edge positioning.
+        Specification is as a `matplotlib.markers` marker, e.g. one of 'so^>v<dph8'.
+
+    label : None or string
+        Label for legend
+
+    min_source_margin : int (default=0)
+        The minimum margin (gap) at the begining of the edge at the source.
+
+    min_target_margin : int (default=0)
+        The minimum margin (gap) at the end of the edge at the target.
+
+    Example
+    -------
+    >>> G = nx.MultiGraph([(0, 1), (0, 1), (0, 2), (0, 2), (0, 2), (3, 0)])
+    >>> positions = {0: (0, 0), 1: (1, -2), 2: (2, 0), 3: (2, 2)}
+    >>> nx.draw_networkx_multigraph_edges(G, pos=positions)
+
+    Notes
+    -----
+    For directed graphs, arrows are drawn at the head end.  Arrows can be
+    turned off by passing an arrowstyle without
+    an arrow on the end.
+
+    Be sure to include `node_size` as a keyword argument; arrows are
+    drawn considering the size of nodes.
+
+    Self-loops are always drawn with `~matplotlib.patches.FancyArrowPatch`
+    regardless of the value of `arrows` or whether `G` is directed.
+    When ``arrows=False`` or ``arrows=None`` and `G` is undirected, the
+    FancyArrowPatches corresponding to the self-loops are not explicitly
+    returned. They should instead be accessed via the ``Axes.patches``
+    attribute (see examples)."""
+
+    if edgelist == None:
+        edgelist = G.edges()
+
+    if arrowstyle == None:
+        if G.is_directed():
+            arrowstyle = "-|>"
+        else:
+            arrowstyle = "-"
+
+    multiEdgeDict = {}
+    for edge in edgelist:
+        if edge not in multiEdgeDict:
+            multiEdgeDict[edge] = 1
+        else:
+            multiEdgeDict[edge] += 1
```

Comment:
I suggest using the method [G.number_of_edges(u, v)](https://networkx.org/documentation/stable/reference/classes/generated/networkx.Graph.number_of_edges.html) to count. Then you can build the dict more easily and also not have to check the opposite direction of the edge.
```python
multiedge_count = {(u, v): G.number_of_edges(u, v) for u, v in G.edges()}
```
Hmmm...  The way I wrote this example computes the number of edges for **each** multiedge -- which is a little silly since they will all be the same value -- and overwrite the previous (identical) entry in the dict. Maybe there is a better way -- but if its not clear don't worry about the extra computes.

Also, just so you know, there is a subtlety that `G.edges()` always yields 2-tuples, while `G.edges` yields 2-tuples for Graph/DiGraph and 3-tuples for MultiGraph/MultiDiGraph.  I don't think it affects things here because you use `G.edges()` for the default above. But it could be that people would include a 3-tuple edge in `edgelist`.  That might be where your use of `e[0]` and `e[1]` will work better than `for u, v in` because it doesn't care if there is a third entry in `e`.

Reply:
If I use `G.number_of_edges` method - instead of counting edges in the edgeList-, would people still be able to use edgelist argument as they expect? For example, a person who prefers to draw exactly one edge between a pair of nodes will not be able to do it if it is a multiedge.

Reply:
That's a good question. Do we want to allow people to choose which edges to draw by listing them in `edgelist`? I think we do.  Do we want to allow them to specify *which* multiedge to draw? Also, probably we do.  So, they should be able to use a 3-tuple in `edgelist` to specify which edges to draw. That means we need to count how many edges in `edgelist` are between the two nodes -- not how many edges in G.edges are between the two nodes...

As for how to count efficiently. I remembered that Python has a collections builtin library with a Counter object. The Counter acts like a dict, only when you construct it or use `count.update(iterable)` it counts how many times each element shows up.  So it is a dict keyed by element to the value that is the number of times that element has been added to the Counter.  I think `from collections import Counter` allows something like: 
```python
edge_counts = Counter(edgelist)
for edge, cnt in edge_counts.items():
    if cnt == 1:
        ...
    else:
        bend = 0.5 / cnt
        ...
```

Reply:
I used Counter as you suggested. However, I think I should still check for opposite direction of the edge because if the graph is directed, there might be an opposite edge even though the edge count is 1.  Consider the following example:

```
G = nx.MultiDiGraph([(0,1),(1,0),(2,0)])
Counter(list(G.edges()))
```

Output: 

`Counter({(0, 1): 1, (1, 0): 1, (2, 0): 1})`

If I don't check for the opposite side, (0, 1) and (1, 0) will be drawn as straight lines and they will overlap. Is there another way to prevent this without checking as I do?

Reply:
Ahh...  hmmm....  I'm not sure how best to handle this case....
Do we bend all the multiedges the same direction (e.g. always to the right?).  

OK...   with multiedges, here's an idea (and maybe there is a better one -- not sure this even works):
- count the number in each direction. Bend them always to the right. Then when the other direction is drawn it will be bent the "other direction" and not overlap. By "other direction" I mean that it still is bent e.g. to the right, but since it is set in the other direction, it will bend away from the edges going the first direction. Does this work?

- In the directed case where we are already drawing FancyArrows, maybe we should bent every edge slightly so that even without multiedges, we still get the opposite direction edges to not overlap. I guess we would want an option to turn that off to get straight line with arrows depicting direction including "both directions".  Hmm.  I guess this can already be done by the user specifying an arc ConnectionStyle for every edge.  maybe??  Not sure...